pymeasure/instruments/rohdeschwarz/hmp.py

```diff
@@ -0,0 +1,271 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from typing import List
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import (strict_discrete_set,
+                                              truncated_range)
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+def process_sequence(sequence: List[float]) -> str:
```

Comment:
We [prefer](https://pymeasure.readthedocs.io/en/latest/dev/coding_standards.html#python-style-guides) (probably mainly for historical reasons) to put type hints into the (sphinx-style, like e.g. [here](https://github.com/pymeasure/pymeasure/blob/8086835d8a211268a26451c28a054770b3907d40/pymeasure/instruments/keithley/keithley2400.py#L387)) docstrings, not the signature/code.
Also in `set_channel_state`

Reply:
Moved the type hints to the docstring.

I think, however, that using the built-in type hints should be allowed. in the future. pymeasure is now Python 3.7+, they are less verbose and easier to read in plain-text format than sphinx-style docstrings and can be checked by your IDE. There is also a plugin for sphinx to make use of type hints: https://pypi.org/project/sphinx-autodoc-typehints/

Reply:
Good to know, thanks!